Marlin/src/sd/cardreader.cpp

```diff
@@ -726,17 +729,20 @@ int8_t CardReader::updir() {
       presort();
     #endif
   }
+  if (!workDirDepth) flag.workDirIsRoot = true;
+  getnrfilenames();
+  getWorkDirName();
```

Comment:
When using SRAM caching with `SDCARD_SORT_ALPHA` the `getnrfilenames` call is managed through the `presort` method. And there's no need to copy the work dir name into the `filename` field, as there is no such expectation and no code is assuming this to be the case.

Reply:
Dear @thinkyhead, removing this lines is OK if we use SDCARD_SORT_ALPHA. but if we dont use SDCARD_SORT_ALPHA, it will create display problem if user touch updir and early page, which will be fixed automatically when ui.firstpage is reached.